symengine/polys/basic_conversions.h

```diff
@@ -0,0 +1,221 @@
+                tmp = div(it, genpow);
+                if (is_a<const Integer>(*tmp)) {
+                    RCP<const Integer> i = rcp_static_cast<const Integer>(tmp);
+                    if (i->is_positive()) {
+                        powr = i->as_int();
+                        continue;
+                    }
+                }
+                coef = mul(coef, pow(genbase, it));
+            }
+            d_add_term(powr, *coef);
+
+        } else {
+            d_add_term(0, x);
+        }
+    }
+
+    void bvisit(const Add &x)
+    {
+        dict = _basic_to_upoly<D, P>(x.coef_, gen);
+        for (auto const &it : x.dict_)
+            dict += (_basic_to_upoly<D, P>(it.first, gen)
+                     * _basic_to_upoly<D, P>(it.second, gen));
+    }
+
+    void bvisit(const Mul &x)
+    {
+        dict = _basic_to_upoly<D, P>(x.coef_, gen);
+        for (auto const &it : x.dict_)
+            dict *= _basic_to_upoly<D, P>(pow(it.first, it.second), gen);
+    }
+
+    void bvisit(const Integer &x)
+    {
+        integer_class i = x.i;
+        if (i != 0)
+            dict = P::container_from_dict(gen, {{0, i}});
+    }
+
+    void bvisit(const Basic &x)
+    {
+        RCP<const Basic> genpow = one, genbase = gen, powr;
+        if (is_a<const Pow>(*gen)) {
+            genpow = rcp_static_cast<const Pow>(gen)->get_exp();
+            genbase = rcp_static_cast<const Pow>(gen)->get_base();
+        }
+
+        if (eq(*genbase, x)) {
+            powr = div(one, genpow);
+            if (is_a<const Integer>(*powr)) {
+                int i = rcp_static_cast<const Integer>(powr)->as_int();
+                if (i > 0) {
+                    dict = P::container_from_dict(
+                        gen, {{i, typename P::coef_type(1)}});
+                    return;
+                }
+            }
+        }
+
+        d_add_term(0, x);
+    }
+};
+
+template <typename Poly>
+class BasicToUIntPoly : public BasicToUPolyBase<Poly, BasicToUIntPoly<Poly>>
+{
+public:
+    using BasicToUPolyBase<Poly, BasicToUIntPoly>::bvisit;
+    using BasicToUPolyBase<Poly, BasicToUIntPoly>::apply;
+
+    void bvisit(const Rational &x)
+    {
+        throw std::runtime_error("Non-integer found");
+    }
+
+    void d_add_term(unsigned int pow, const Basic &x)
+    {
+        if (is_a<const Integer>(x))
+            this->dict = Poly::container_from_dict(
+                this->gen, {{pow, static_cast<const Integer &>(x).i}});
+        else
+            throw std::runtime_error("Non-integer found");
+    }
+};
+
+class BasicToUExprPoly : public BasicToUPolyBase<UExprPoly, BasicToUExprPoly>
+{
+public:
+    using BasicToUPolyBase<UExprPoly, BasicToUExprPoly>::bvisit;
+    using BasicToUPolyBase<UExprPoly, BasicToUExprPoly>::apply;
+
+    void bvisit(const Rational &x)
+    {
+        dict = UExprDict(x.rcp_from_this());
+    }
+
+    void d_add_term(unsigned int pow, const Basic &x)
+    {
+        dict = UExprDict({{pow, x.rcp_from_this()}});
```

Comment:
I initially thought `d_add_term` meant adding a term to a dictionary, but it's not. Can you change the name?